mcs/class/System.IO.Compression/Test/System.IO.Compression/ZipTest.cs

```diff
@@ -517,5 +522,16 @@ public void ZipReadNonSeekableStream()
 			{
 			}
 		}
+
+		[Test]
+		public void ZipWriteNonSeekableStream() {
+			var stream = new MyFakeStream( "test.nupkg", FileMode.Open );
+			using ( var archive = new ZipArchive( stream, ZipArchiveMode.Create ) ) {
+				var entry = archive.CreateEntry( "foo" );
+				using ( var es = entry.Open() ) {
+					es.Write( new byte[] { 4, 2 }, 0, 2 );
```

Comment:
Test showing the problem. In Create mode, ZipArchive would previously attempt to set both Position and Length on the stream.